### PR TITLE
Replace browsermob-proxy with browserup-proxy

### PR DESCRIPTION
**Pull request by TatsianaLuchonak**

### CODE REVIEW

File: vividus-plugin-web-app/build.gradle [hunk 1]
```diff
@@ -29,7 +29,8 @@ dependencies {
     implementation(group: 'org.seleniumhq.selenium', name: 'selenium-safari-driver', version: seleniumVersion)
     implementation(group: 'junit', name: 'junit', version: versions.junit4)
     implementation(group: 'io.github.bonigarcia', name: 'webdrivermanager', version: '3.6.2')
-    implementation(group: 'net.lightbody.bmp', name: 'browsermob-core', version: '2.1.5')
+    implementation(group: 'com.browserup', name: 'browserup-proxy-core', version: '2.0.0')
```
Comment (valfirst) [inline vividus-plugin-web-app/build.gradle]: `2.0.1` ?

File: vividus-plugin-web-app/build.gradle [hunk 2]
```diff
@@ -29,7 +29,8 @@ dependencies {
...
+    implementation(group: 'xyz.rogfam', name : 'littleproxy', version : '2.0.0-beta-3')
```
Comment (valfirst) [inline vividus-plugin-web-app/build.gradle]: `2.0.0-beta-5` ?

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/deserializer/HarNameValuePairDeserializer.java [hunk 3]
```diff
@@ -18,22 +18,25 @@
...
+    public HarQueryParam deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
             throws IOException
     {
         JsonNode jsonNode = jsonParser.getCodec().readTree(jsonParser);
         String name = jsonNode.get("name").asText();
         String value = jsonNode.get("value").asText();
-        return new HarNameValuePair(name, value);
+        HarQueryParam harQueryParam = new HarQueryParam();
+        harQueryParam.setName(name);
+        harQueryParam.setValue(value);
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/deserializer/HarNameValuePairDeserializer.java]: it looks like having this new code deserializer is not required

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/deserializer/HarNameVersionDeserializer.java [hunk 4]
```diff
@@ -18,22 +18,25 @@
...
+    public HarCreatorBrowser deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
             throws IOException
     {
         JsonNode jsonNode = jsonParser.getCodec().readTree(jsonParser);
         String name = jsonNode.get("name").asText();
         String version = jsonNode.get("version").asText();
-        return new HarNameVersion(name, version);
+        HarCreatorBrowser harCreatorBrowser = new HarCreatorBrowser();
+        harCreatorBrowser.setName(name);
+        harCreatorBrowser.setVersion(version);
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/deserializer/HarNameVersionDeserializer.java]: it looks like having this new code deserializer is not required

File: vividus-plugin-web-app/build.gradle (hunk 1 above)
Comment (TatsianaLuchonak) [reply vividus-plugin-web-app/build.gradle]: Yes, I set it on Friday, before 2.0.1 release. Will fixed

**Comment by uarlouski** [comment]
What are benefits to use browserup proxy in place of browsermob?

**Comment by TatsianaLuchonak** [comment]
Browserup is fork version of browsermob which supported and released regularly

**Comment by valfirst** [comment]
> What are benefits to use browserup proxy in place of browsermob?

moreover it eliminates some limitations of BMP, see PR details

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/ProxyLog.java [hunk 5]
```diff
@@ -115,7 +116,8 @@ public String getRequestUrl(String urlPattern)
     {
         String httpMethodStr = httpMethod.toString();
         return getFilteredHarEntriesStream(urlPattern)
-                .filter(entry -> httpMethodStr.equals(entry.getRequest().getMethod())).collect(toList());
+                .filter(entry -> httpMethodStr.equals(entry.getRequest().getMethod().toString()))
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/ProxyLog.java]: `entry -> httpMethod == entry.getRequest().getMethod()` ?

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/parser/HarParser.java [hunk 6]
```diff
@@ -20,26 +20,19 @@
...
-import net.lightbody.bmp.core.har.HarNameValuePair;
-import net.lightbody.bmp.core.har.HarNameVersion;
-
 public class HarParser implements IHarParser
 {
     private final ObjectMapper objectMapper;
 
     public HarParser(List<DeserializationProblemHandler> deserializationProblemHandlers)
     {
         SimpleModule module = new SimpleModule();
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/parser/HarParser.java]: so `module` is not needed anymore, is it?

**Pull request merged**